src/wp-includes/blocks/navigation.php

```diff
@@ -1442,13 +1441,12 @@ function block_core_navigation_insert_hooked_blocks( $inner_blocks, $post ) {
  */
 function block_core_navigation_set_ignored_hooked_blocks_metadata( $inner_blocks, $post ) {
 	$mock_navigation_block = block_core_navigation_mock_parsed_block( $inner_blocks, $post );
-	$hooked_blocks         = get_hooked_blocks();
 	$before_block_visitor  = null;
 	$after_block_visitor   = null;
 
-	if ( ! empty( $hooked_blocks ) || has_filter( 'hooked_block_types' ) ) {
-		$before_block_visitor = make_before_block_visitor( $hooked_blocks, $post, 'set_ignored_hooked_blocks_metadata' );
-		$after_block_visitor  = make_after_block_visitor( $hooked_blocks, $post, 'set_ignored_hooked_blocks_metadata' );
+	if ( maybe_has_hooked_blocks() ) {
```

Comment:
This code comes from Gutenberg, so we would need to update it in GB also if we proceed.